plugins/@grouparoo/iterable/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,101 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import { addToList, removeFromList } from "./listMethods";
+
+export const exportProfile: ExportProfilePluginMethod = async (args) => {
+  try {
+    return sendProfile(args);
+  } catch (error) {
+    if (error?.response?.status === 429) {
+      return { error, success: false, retryDelay: 1000 * 11 }; // the most common rate-limit error from hubspot is in 10-second intervals
+    } else {
+      throw error;
+    }
+  }
+};
+
+export const sendProfile: ExportProfilePluginMethod = async ({
+  appGuid,
+  appOptions,
+  destinationOptions,
+  export: {
+    toDelete,
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+
+  const email = newProfileProperties["email"]; // this is how we will identify profiles
+  const currentEmail = oldProfileProperties["email"];
+  if (!email) {
+    throw new Error(`newProfileProperties[email] is a required mapping`);
+  }
+
+  if (toDelete) {
+    await client.users.delete(email);
```

Comment:
note, it could also be the old user to delete.

Reply:
can you explain better?

Reply:
The user that needs to be deleted could be `currentEmail` from `oldProperties`. If you went with the code I referenced in https://github.com/grouparoo/grouparoo/pull/1204#discussion_r566486570, then you'd maybe add deleting to that concept or do something like that to see who is really in there.